compiler/src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -207,11 +207,16 @@ abstract class Reporter extends interfaces.ReporterResult {
       b += countString(warningCount, "warning") + " found"
     if (errorCount > 0)
       b += countString(errorCount, "error") + " found"
-    for ((settingName, count) <- unreportedWarnings)
-      b += s"there were $count ${settingName.tail} warning(s); re-run with $settingName for details"
     b.mkString("\n")
   }
 
+  def summarizeUnreportedWarnings(using Context): Unit =
+    val b = mutable.ListBuffer[String]()
+    for (settingName, count) <- unreportedWarnings do
+      b += s"there were $count ${settingName.tail} warning(s); re-run with $settingName for details"
+    val s = b.mkString("\n")
+    if s.nonEmpty then report(Info(s, NoSourcePosition))
```

Comment:
```
[warn] two warnings found
[info] there were 3 deprecation warning(s); re-run with -deprecation for details
```
The report should probably be a warning instead of an info.

I don't understand why the `report(new Info(s, NoSourcePosition))`  on line 223, in printSummary, appears as a warning.

Reply:
> I don't understand why the `report(new Info(s, NoSourcePosition))` on line 223, in `printSummary`, appears as a warning.

https://github.com/sbt/zinc/blob/fe53448eed5422809b0fe464d314378e89419932/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/LoggedReporter.scala#L151

Reply:
I've updated the behavior to be consistent with Scala 2, namely the unreported warning summaries are reported

  - with a level of warning rather than info
  - individually rather than as one newline delimited report
  - before the summary of warning/error counts, so they are included in the warning count

This required updating a few tests and check files.